modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidgetEnhanced.php

```diff
@@ -0,0 +1,639 @@
+<?php
+
+namespace Drupal\tide_content_collection\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextareaWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\tide_automated_listing\SearchApiIndexHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Implementation of the content collection configuration enhanced widget.
+ *
+ * @FieldWidget(
+ *   id = "content_collection_configuration_enhanced",
+ *   label = @Translation("Content Collection Configuration (Enhanced)"),
+ *   field_types = {
+ *     "content_collection_configuration"
+ *   }
+ * )
+ */
+class ContentCollectionConfigurationWidgetEnhanced extends StringTextareaWidget implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Search API Index helper.
+   *
+   * @var \Drupal\tide_automated_listing\SearchApiIndexHelper
+   */
+  protected $indexHelper;
+
+  /**
+   * The module handler service.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The search API index.
+   *
+   * @var \Drupal\search_api\IndexInterface
+   */
+  protected $index;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, ModuleHandlerInterface $module_handler, SearchApiIndexHelper $index_helper) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->moduleHandler = $module_handler;
+    $this->indexHelper = $index_helper;
+    $this->getIndex();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('module_handler'),
+      $container->get('tide_automated_listing.sapi_index_helper')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'schema_validation' => FALSE,
+      'callToAction' => FALSE,
+      'contentTypes' => TRUE,
+      'field_topic' => TRUE,
+      'field_tags' => TRUE,
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() : array {
+    $summary = parent::settingsSummary();
+
+    $schema_validation = $this->getSetting('schema_validation');
```

Comment:
@joshua-salsadigital from the SDs:
> The new widget should not validate against the JSON schema.

Reply:
Reviewed the SD and made changes as per the SD